### PR TITLE
fix: blob add

### PR DESCRIPTION
**Pull request by Gozala**
Fixes various issues to get tests passing

### CODE REVIEW

File: packages/upload-api/src/blob/add.js [hunk 1]
```diff
@@ -161,6 +161,11 @@ async function allocate({ context, blob, space, cause }) {
 
   // 3. if not already allocated (or expired) execute `blob/allocate`
   if (!blobAllocateReceipt) {
+    // Create allocation task and save it
+    const saveTask = await context.tasksStorage.put(task)
+    if (!saveTask.ok) {
+      return saveTask
+    }
```
Comment (Gozala) [inline packages/upload-api/src/blob/add.js:168]: `ucan/conclude` seems to assume that `blob/allocate` task is stored here, but nothing was storing it which caused errors when `http/put` receipts were submitted. I'm not sure what @vasco-santos had intended here, as some tests seemed to manually store these. But then some handlers seemed to save task from the handler.

I think it is reasonable to save task from the handler especially if we intend to look those up later.

File: packages/upload-client/src/blob.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ export async function add(
   const bytes =
     car instanceof Uint8Array ? car : new Uint8Array(await car.arrayBuffer())
   const link = await CAR.codec.link(bytes)
-  const digest = link.bytes
+  const digest = link.multihash.bytes
```
Comment (Gozala) [inline packages/upload-client/src/blob.js:176]: @joaosa this was the culprit that caused test failures, link is a CID which is CID prefix + multihash which is why errors where saying digest length was greater than expected.

File: packages/upload-client/src/blob.js [hunk 3]
```diff
@@ -279,30 +279,10 @@ export async function add(
...
-      },
-      headers: {
-        'ucan/await': ['.out.ok.address.headers', nextTasks.allocate.task.cid],
-      },
-    },
-    facts: nextTasks.put.task.facts,
-    expiration: Infinity,
-  })
 
-  const httpPutDelegation = await httpPut.delegate()
```
Comment (Gozala) [inline packages/upload-client/src/blob.js:302]: I removed this because it effectively recreates same task that we already have reference to.

**Review by joaosa — APPROVED** [review]
LGTM

**Pull request merged**